src/utils/config_filters.cpp

```diff
@@ -71,7 +71,7 @@ bool utils::config_filters::unsigned_matches_if_present(const config& filter, co
 		return false;
 	}
 
-	return in_ranges<int>(cfg[attribute].to_int(0), utils::parse_ranges_unsigned(filter[attribute].str()));
+	return in_ranges<unsigned>(cfg[attribute].to_unsigned(0), utils::parse_ranges_unsigned(filter[attribute].str()));
```

Comment:
This seems wrong? Attributes can have negative values.

Reply:
Would that mean the original code was buggy? Because a negative value would always fail regardless.

Reply:
Well, I suppose it doesn't matter either way given the function is unused:
```
grep -Iirn 'unsigned_matches_if_present'
utils/config_filters.cpp:65:bool utils::config_filters::unsigned_matches_if_present(const config& filter, const config& cfg, const std::string& attribute)
utils/config_filters.hpp:52:bool unsigned_matches_if_present(const config& filter, const config& cfg, const std::string& attribute);
tests/test_config_filters.cpp:31:       // but for the unsigned_matches_if_present function only one of them is
tests/test_config_filters.cpp:36:       BOOST_ASSERT(!unsigned_matches_if_present(minus_3, minus_3, "x"));
tests/test_config_filters.cpp:37:       BOOST_ASSERT(unsigned_matches_if_present(plus_3, plus_3, "x"));
tests/test_config_filters.cpp:39:       BOOST_ASSERT(!unsigned_matches_if_present(minus_3, plus_3, "x"));
tests/test_config_filters.cpp:40:       BOOST_ASSERT(!unsigned_matches_if_present(plus_3, minus_3, "x"));
```
I think it's worth considering what would happen in other cases though if the value did actually underflow to the max value of `unsigned`.

Reply:
That does make me realize one small hole in these changes (and I guess the original implementation). `infinity` is represented as the max value of `int`... which is half the max value of `unsigned`. 🤔 One would expect parsing an unsigned range to infinity would include that extra space.